packages/react-devtools-shared/src/devtools/views/Components/ComponentsResizer.js

```diff
@@ -0,0 +1,246 @@
+  horizontalPercentage: number,
+  isResizing: boolean,
+  verticalPercentage: number,
+|};
+
+function initResizeState(): ResizeState {
+  let horizontalPercentage = 0.65;
+  let verticalPercentage = 0.5;
+
+  try {
+    let data = localStorageGetItem(LOCAL_STORAGE_KEY);
+    if (data != null) {
+      data = JSON.parse(data);
+      horizontalPercentage = data.horizontalPercentage;
+      verticalPercentage = data.verticalPercentage;
+    }
+  } catch (error) {}
+
+  return {
+    horizontalPercentage,
+    isResizing: false,
+    verticalPercentage,
+  };
+}
+
+function resizeReducer(state: ResizeState, action: ResizeAction): ResizeState {
+  switch (action.type) {
+    case 'ACTION_SET_IS_RESIZING':
+      return {
+        ...state,
+        isResizing: action.payload,
+      };
+    case 'ACTION_SET_HORIZONTAL_PERCENTAGE':
+      return {
+        ...state,
+        horizontalPercentage: action.payload,
+      };
+    case 'ACTION_SET_VERTICAL_PERCENTAGE':
+      return {
+        ...state,
+        verticalPercentage: action.payload,
+      };
+    default:
+      return state;
+  }
+}
+
+function getOrientation(
+  wrapperElementRef: ElementRef<HTMLElement>,
+): null | Orientation {
+  const wrapperElement = wrapperElementRef.current;
+  if (wrapperElement != null) {
+    const {width} = wrapperElement.getBoundingClientRect();
+    return width > VERTICAL_MODE_MAX_WIDTH ? 'horizontal' : 'vertical';
+  }
+  return null;
+}
+
+function setResizeCSSVariable(
+  resizeElementRef: ElementRef<HTMLElement>,
+  orientation: null | Orientation,
+  percentage: number,
+): void {
+  const resizeElement = resizeElementRef.current;
+
+  if (resizeElement !== null && orientation !== null) {
+    resizeElement.style.setProperty(
+      `--${orientation}-resize-percentage`,
+      `${percentage}%`,
+    );
+  }
+}
+
+function createResizeReducer(wrapperElementRef, resizeElementRef) {
+  const [state, dispatch] = useReducer<ResizeState, ResizeAction>(
+    resizeReducer,
+    null,
+    initResizeState,
+  );
+  const {horizontalPercentage, verticalPercentage} = state;
+  const orientationRef = useRef(null);
+
+  // Initial set up for the resize percentage CSS variables.
+  useLayoutEffect(() => {
+    const orientationAndPercentage = {
+      horizontal: horizontalPercentage,
+      vertical: verticalPercentage,
+    };
+
+    Object.keys(orientationAndPercentage).forEach(orientation => {
+      setResizeCSSVariable(
+        resizeElementRef,
+        orientation,
+        orientationAndPercentage[orientation] * 100,
+      );
+    });
+  }, []);
+
+  useLayoutEffect(() => {
+    const orientation = getOrientation(wrapperElementRef);
+
+    if (orientation !== orientationRef.current) {
+      orientationRef.current = orientation;
+
+      const percentage =
+        orientation === 'horizontal'
+          ? horizontalPercentage
+          : verticalPercentage;
+
+      setResizeCSSVariable(resizeElementRef, orientation, percentage * 100);
+    }
+  });
```

Comment:
This could be a mount effect only, I think. No need for this to run as a layout effect after every update. :smile:

```suggestion
  useLayoutEffect(() => {
    setResizeCSSVariable(
      resizeElementRef,
      'horizontal',
      horizontalPercentage * 100,
    );
    setResizeCSSVariable(
      resizeElementRef,
      'vertical',
      verticalPercentage * 100,
    );
  }, []);
```